apis/nucleus/src/components/Cell.jsx

```diff
@@ -282,12 +283,32 @@ const Cell = forwardRef(
     const [state, dispatch] = useReducer(contentReducer, initialState(initialError));
     const [layout, { validating, canCancel, canRetry }, longrunning] = useLayout(model);
     const [appLayout] = useAppLayout(app);
-    const [contentRef, contentRect] = useRect();
+    const [contentRef, contentRect, contentNode] = useRect();
     const [snOptions, setSnOptions] = useState(initialSnOptions);
     const [snPlugins, setSnPlugins] = useState(initialSnPlugins);
     const [selections] = useObjectSelections(app, model);
     const [hovering, setHover] = useState(false);
     const hoveringDebouncer = useRef({ enter: null, leave: null });
+    const focusHandler = useRef({
+      focusToolbarButton(last) {
+        // eslint-disable-next-line react/no-this-in-sfc
+        this.emit(last ? 'focus_toolbar_last' : 'focus_toolbar_first');
+      },
+    });
+
+    useEffect(() => {
+      eventmixin(focusHandler.current);
+    }, []);
+
+    focusHandler.current.blurCallback = (resetFocus) => {
+      halo.root.toggleFocusOfCells();
+      if (resetFocus && contentNode) {
+        contentNode.focus();
+      }
+    };
+    focusHandler.current.refocusContent = () => {
+      typeof state.sn.component?.focus === 'function' && state.sn.component.focus();
+    };
```

Comment:
So the `focusHandler` instance need to stay the same, and `eventmixin` should only run the first time. The other methods I need to add afterwards. Not sure why but it worked this way